### ISSUE TITLE
test_import_without_warning fails with packaging >= 20.5

### ISSUE THREAD

**Issue by bnavigator** [dask/fastparquet#558]
`test_import_without_warning` does this:

```
subprocess.check_call([sys.executable, "-Werror", "-c", "import fastparquet"])
```

But with packaging >= 20.5 this raises a DeprecationWarning:
```
[   95s] E           subprocess.CalledProcessError: Command '['/usr/bin/python3.6', '-Werror', '-c', 'import fastparquet']' returned non-zero exit status 1.
...
[   95s] DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
```

This is due to https://github.com/pypa/setuptools/issues/2466 / https://github.com/pypa/packaging/issues/368

**Comment by martindurant** [comment]
If you're willing, this sounds like it should be easily fixable

**Comment by bnavigator** [comment]
The easy fix is to just ignore the test (`pytest -k "not  test_import_without_warning"` or a decorator for the test `@pytest.mark.skipif(packaging.__version__ >= "20.5")`). The full fix is in the hands of pypa.

**Comment by martindurant** [comment]
Is it the use of `LooseVersion` that triggers this? I think its couple of instances can be easily dropped.

**Comment by bnavigator** [comment]
No, it is completely internal to `pkg_resources`. Nothing to be done in fastparquet. I stepped through it with `pdb.run('import fastparquet')`. The problem is the same as in https://github.com/pypa/setuptools/issues/2466#issuecomment-738731493: `parts` starts with `fastparquet`, which is not a valid version.

**Comment by martindurant** [comment]
That is somewhat mystifying. Can we reformat the version? I don't mind dropping the test too much, but I don't want py39 users to have to see this warning.

**Comment by bnavigator** [comment]
I run into this on Python 3.8. It's the newer packaging.

We can't  reformat the version because it fails even before:
https://github.com/pypa/setuptools/blob/c121d289da5d19cf6df2bf6b64ac28916a060161/pkg_resources/__init__.py#L2025
```
        name, ext = os.path.splitext(name)
        parts = itertools.chain(name.split('-'), [ext])
        return [packaging.version.parse(part) for part in parts]
```
name is `fastparquet-0.5.0-py3.8` which is valid and expected.

**Comment by bnavigator** [comment]
> but I don't want py39 users to have to see this warning.

End-Users normally don't see `DeprecationWarning`s. Only the -Werror makes it visible and throws the Exception.

**Comment by martindurant** [comment]
OK then, just skip the test, or make it not worry about this kind of warning!
